pkg/database/prepared.go

```diff
@@ -0,0 +1,170 @@
+package database
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"sync"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const otelName = "prepared"
+
+var otelTracer trace.Tracer
+
+func init() {
+	otelTracer = otel.Tracer(otelName)
+}
+
+type preparedDb[T Transaction] struct {
+	underlying Wrapped[T]
+	mu         sync.RWMutex
+	store      map[string]*sql.Stmt
```

Comment:
```suggestion
	cache      map[string]*sql.Stmt
```
Consider renaming store to cache?